draft-ietf-quic-tls.md

```diff
@@ -1059,6 +1059,27 @@ could either be absent or it could indicate that an address validation token is
 not present.
 
 
+### Stateless Address Validation
+
+A server can use the cookie extension to store all state necessary to continue
+the connection.  This allows a server to avoid committing state for clients that
+have unvalidated source addresses.
+
+For instance, a server could use a statically-configured key to encrypt the
+information that it requires and include that information in the cookie.  In
+addition to address validation information, a server that uses encryption also
+needs to be able recover the hash of the ClientHello and its length, plus any
+information it needs in order to reconstruct the HelloRetryRequest.
+
+A client proceeds as though the server were stateful.  It constructs a second
+ClientHello and sends it following the initial ClientHello.  Note that this
+means a server that doesn't maintain state between the intial and second
+ClientHello will receive the second ClientHello in STREAM frames with a non-zero
+offset.  The resulting gap in the received stream data needs to be ignored and
```

Comment:
@abraud is correct.  What makes this hard is that the server might have generated multiple HelloRetryRequest packets, each with their own randomized packet number.  That's a real issue because it means that it can't save that value without changing the HelloRetryRequest.  Then the client will receive two conflicting bits of stream data.  That's bad because we allow endpoints to reject connections if that happens.  Separately, @janaiyengar suggests that we treat a stateless reject as a clean break.  I will see if I can use that; it will need a lot more text :(

@mikkelfj, you can't send anything other than a STREAM frame.

@RyanAtGoogle, you got it right.  The server saves the offset, then pretends that it received those missing bytes once it has validated the token.  If we take @janaiyengar's suggestion though, that's won't be necessary.